blocks/aem-asset-selector/aem-asset-selector-util.js

```diff
@@ -55,62 +155,215 @@ export function init(cfg, callback) {
+ *  copied successfully, otherwise resolves with false.
+ */
+export async function copyAssetWithoutRapi(asset) {
+  const maxRendition = getCopyRendition(asset);
+  if (!maxRendition) {
+    logMessage('No suitable rendition to copy found');
+    return false;
+  }
+  try {
+    const assetPublicUrl = await getAssetPublicUrl(maxRendition.href.substring(0, maxRendition.href.indexOf('?')));
+    if (!assetPublicUrl) {
+      logMessage('Unable to generate public url for copy');
+      return false;
+    }
+    await copyToClipboardWithHtml(assetPublicUrl);
+  } catch (e) {
+    logMessage('error copying asset to clipboard', e);
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Copies the given asset to the clipboard, using the Repository API to
+ * generate a download URL for the asset, then using the generated URL
+ * to copy the asset's content to the clipboard.
+ * @param {Asset} asset Asset repository metadata, as provided by the
+ *  asset selector.
+ * @returns {Promise<boolean>} Resolves with true if the asset was
+ *  copied successfully, otherwise resolves with false.
+ */
+export async function copyAssetWithRapi(asset) {
+  // eslint-disable-next-line no-underscore-dangle
+  if (!asset) {
+    logMessage('Asset metadata does not contain sufficient information');
+    return false;
+  }
+  const rendition = getCopyRendition(asset);
+  if (!rendition) {
+    logMessage('No suitable rendition to copy found');
+    return false;
+  }
+  const download = getRel(rendition, REL_DOWNLOAD);
+  if (!download || !download.href) {
+    logMessage('Rendition does not contain sufficient information');
+    return false;
+  }
+  try {
+    const url = download.href;
+    const res = await fetch(url, {
+      headers: {
+        Authorization: `Bearer ${imsInstance.getImsToken()}`,
+      },
+    });
+    if (!res.ok) {
+      logMessage(`Download request for rendition binary returned unexpected status code ${res.status}: ${res.statusText}`);
+      return false;
+    }
+    const downloadJson = await res.json();
+    await copyToClipboardWithBinary(downloadJson.href, downloadJson.type);
+  } catch (e) {
+    logMessage('error copying asset to clipboard', e);
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Retrieves the asset selector's containing element.
+ * @returns {HTMLElement} The asset selector.
+ */
+function getAssetSelector() {
+  return document.getElementById('asset-selector');
+}
+
+/**
+ * Invoked when the currently selected asset in the asset selector has
+ * changed. Will invoke either onAssetSelected() or onAssetDeselected()
+ * depending on the new selection state.
+ * @param {Array<Asset>} selection The new selection in the selector.
+ * @param {AssetSelectorConfig} cfg Configuration for the asset selector.
+ */
+function handleAssetSelection(selection, cfg) {
+  if (cfg) {
+    if (selection.length && cfg.onAssetSelected) {
+      cfg.onAssetSelected(selection[0]);
```

Comment:
If length > 1, can we log that we use only the first one? Or at least explain it in the readme.